src/test/java/com/vackosar/gitflowincrementalbuild/control/DifferentFilesTest.java

```diff
@@ -230,59 +178,30 @@ public void fetchNonExistent() throws Exception {
         Assert.assertEquals(FETCH_FILE, remoteGit.log().setMaxCount(1).call().iterator().next().getFullMessage());
-        Property.fetchReferenceBranch.setValue(Boolean.TRUE.toString());
-        Property.referenceBranch.setValue(REMOTE_DEVELOP);
-        getInstance(localRepoMock.getBaseCanonicalBaseFolder().toPath()).get();
+        projectProperties.setProperty(Property.fetchReferenceBranch.fullName(), "true");
+        projectProperties.setProperty(Property.referenceBranch.fullName(), REMOTE_DEVELOP);
+
+        invokeUnderTest();
+
         localGit.reset().setMode(ResetCommand.ResetType.HARD).call();
         localGit.checkout().setName(REMOTE_DEVELOP).call();
         Assert.assertEquals(FETCH_FILE, localGit.log().setMaxCount(1).call().iterator().next().getFullMessage());
     }
 
-    private static class ModuleFacade extends AbstractModule {
-        private final GuiceModule guiceModule;
-        private Git git;
-        private Path workDir;
-
-        public ModuleFacade(Path dir) throws Exception {
-            this.guiceModule = new GuiceModule(new ConsoleLogger(), MavenSessionMock.get(dir));
-            this.workDir = dir;
-        }
-
-        @Singleton @Provides public Logger provideLogger() {
-            return new ConsoleLoggerManager().getLoggerForComponent("Test");
-        }
-
-        @Singleton @Provides public Git provideGit(Configuration configuration) throws IOException, GitAPIException {
-            git = guiceModule.provideGit(new StaticLoggerBinder(new ConsoleLoggerManager().getLoggerForComponent("Test")), configuration);
-            return git;
-        }
-
-        @Singleton @Provides public Configuration configuration() throws Exception {
-            MavenSession mavenSession = MavenSessionMock.get(workDir);
-            return new Configuration(mavenSession);
-        }
-
-        @Override
-        protected void configure() {}
-
-        public void close() {
-            if (git != null) {
-                if (git.getRepository() != null) {
-                    git.getRepository().close();
-                }
-                git.close();
-            }
-        }
-
+    private Set<Path> invokeUnderTest() throws Exception {
+        return invokeUnderTest(getMavenSessionMock());
     }
 
-    private DifferentFiles getInstance(Path dir) throws Exception {
-        moduleFacade = new ModuleFacade(dir);
-        return Guice.createInjector(moduleFacade).getInstance(DifferentFiles.class);
-    }
+    private Set<Path> invokeUnderTest(final MavenSession mavenSessionMock) throws Exception {
+        mavenSessionMock.getTopLevelProject().getProperties().putAll(projectProperties);
+
+        DifferentFiles underTest = new DifferentFiles();
+        Whitebox.setInternalState(underTest, mavenSessionMock, CONSOLE_LOGGER);
+
+        Set<Path> result = underTest.get(new Configuration(mavenSessionMock));
 
-    private void setWorkDir(final Path path) {
-        System.setProperty("user.dir", path.toString());
```

Comment:
Note: I removed this entirely because it didn't seem to anything useful for the test. It was also only set for a fraction of the test methods.